objects.md

```diff
@@ -0,0 +1,118 @@
+# About
+
+This repo needs to represent the following information:
+
+1. Adding/removing versions to channels
+2. Remove or add edges to all channels
+3. Remove or add edges to a specific channel
+4. Implement an "upgrade policy" object specifically for y-streams
+
+Remember that cininnatti should pull most edges from the release manifest. This data should typiclly remove edges, not add them.
+
+# Directory Structure
+
+We do this via a combination of objects and directory structure. All object should be YAML (so we can make comments). The directories in the repo should be:
+- /nodes/
+- /channels/
+- /added_edges/
+- /blocked_edges/
+- /y-policies/
+
+# Nodes
+
+The /nodes objects are very simple. It is not strictly necessary as long as cincinnatti pulls all releases from quay. However it is useful for a CI tool to build a final (smaller) graph and for disconnected customers to build a complete cincinnatti graph. (though the later is not the first target goal!) So /nodes/4.1.24 would look like:
+```
+version: 4.1.24
+payload: quay.io/openshift-release-dev/ocp-release@sha256:6f87fb66dfa907db03981e69474ea3069deab66358c18d965f6331bd727ff23f
+```
+
+# Channels
+
+The /channels object described the nodes in a channel. For example the /channels/stable-4.1.yaml file would contain
+
+```
+name: stable-4.1
+versions:
+  - 4.1.0
+  - 4.1.1
+  - ...
+  - 4.1.11
+  # 4.1.12 Was excluded from stable!!!
+  - 4.1.13
+  - 4.1.14
+subchannels:
+```
+
+Subchannels can be implemented later. The idea is that inside stable-4.2.yaml we would have:
+```
+subchannels:
+  - stable-4.1
+```
+Which would pull all nodes from stable-4.1 into stable-4.2. Without having to explicitly add them.
```

Comment:
Can we drop this business about subchannels?  I think we want to restrict `stable-4.2` tagging to 4.1 releases that we (at one point) felt were good jumping-off points for 4.1->4.2 updates.  I don't see a need to tag older 4.1 releases into 4.2 and would prefer other approaches for "I'm on some ancient 4.1.z and want to get over to 4.2".  But we can sort that all out somewhere else without blocking a Git flow for our current models if we drop the subchannel references here.

Reply:
agree that subchannels are not required in an initial implementation. But I'm not convinced to give up on the idea...